src/main/java/com/powsybl/openloadflow/equations/EquationTerm.java

```diff
@@ -103,6 +106,76 @@ static EquationTerm multiply(EquationTerm term, double scalar) {
         return new MultiplyByScalarEquationTerm(term, scalar);
     }
 
+    class VariableEquationTerm extends AbstractEquationTerm {
+
+        private final int elementNum;
+
+        private final List<Variable> variables;
+
+        private double value;
+
+        VariableEquationTerm(int elementNum, VariableType variableType, VariableSet variableSet) {
+            this.elementNum = elementNum;
+            this.variables = Collections.singletonList(variableSet.getVariable(elementNum, variableType));
+        }
+
+        @Override
+        public ElementType getElementType() {
+            return variables.get(0).getType().getElementType();
+        }
+
+        @Override
+        public int getElementNum() {
+            return elementNum;
+        }
+
+        @Override
+        public List<Variable> getVariables() {
+            return variables;
+        }
+
+        @Override
+        public void update(double[] x) {
+            value = x[variables.get(0).getRow()];
+        }
+
+        @Override
+        public double eval() {
+            return value;
+        }
+
+        @Override
+        public double der(Variable variable) {
+            return 1;
+        }
+
+        @Override
+        public boolean hasRhs() {
+            return false;
+        }
+
+        @Override
+        public double rhs() {
+            return 0;
+        }
+
+        @Override
+        public void write(Writer writer) throws IOException {
+            variables.get(0).write(writer);
+        }
+    }
+
+    static VariableEquationTerm createVariable(LfElement element, VariableType variableType, VariableSet variableSet) {
```

Comment:
The name of the method is really confusing for me. I think that we have to be totally explicit: `createVariableEquationTerm`?

Reply:
```java
EquationTerm.createVariableEquationTerm
```
would be redundant ?

Reply:
`createVariableTerm`?

Reply:
Done!